definitions/marts/looker_studio_marts/ecf/ls_ecf_correct_cohort_checker.sqlx

```diff
@@ -0,0 +1,255 @@
+    -- Calculate earliest declaration date for the participant for funded declarations only.
+    ,MIN(CASE WHEN state IN ('paid', 'payable', 'eligible') THEN DATE(declaration_date) ELSE DATE(2050,12,31) END) OVER (PARTITION BY participant_profile_id) AS earliest_declaration_date
+  FROM
+    ${ref('ls_declarations_provider_names')}
+  WHERE
+    programme = 'ECF'
+    AND
+    state IN (
+       'paid'
+      ,'payable'
+      ,'eligible'
+      ,'submitted'
+    )
+),
+
+-- Join declarations onto deduped induction records & add on the grouping of suggested cohorts by induction start date and earliest declaration..
+combined_data AS (
+  SELECT 
+     pro.user_id
+    ,pro.participant_profile_id
+    ,pro.participant_role
+    ,pro.schedule_identifier AS current_schedule
+    ,dec.declaration_id
+    ,dec.declaration_date
+    ,dec.lead_provider_name
+    ,IF(pro.induction_lead_provider_name = 'No lead provider', NULL, pro.induction_lead_provider_name) AS induction_lead_provider_name
+    ,dec.declaration_type
+    ,dec.declaration_state
+    ,pro.cohort AS current_cohort
+    ,pro.induction_start_date
+    ,pro.eligible_for_funding
+    ,pro.partnership_id
+    ,pro.partnership_relationship
+    ,dec.earliest_declaration_date AS earliest_declaration_date
+    ,CASE
+      WHEN induction_start_date < '2021-09-01' THEN '2023'
+      WHEN induction_start_date BETWEEN '2021-09-01' AND '2022-05-31' THEN '2021'
+      WHEN induction_start_date BETWEEN '2022-06-01' AND '2022-08-31' THEN '2021,2022'
+      WHEN induction_start_date BETWEEN '2022-09-01' AND '2023-05-31' THEN '2022'
+      WHEN induction_start_date BETWEEN '2023-06-01' AND '2023-08-31' THEN '2022,2023'
+      WHEN induction_start_date BETWEEN '2023-09-01' AND '2024-05-31' THEN '2023'
+      ELSE NULL
+    END AS start_date_suggested_true_cohort
+    ,CASE
+      WHEN earliest_declaration_date BETWEEN '2021-09-01' AND '2023-05-31' THEN '2021,2022'
+      WHEN earliest_declaration_date BETWEEN '2023-06-01' AND '2023-08-31' THEN '2021,2022,2023'
+      WHEN earliest_declaration_date BETWEEN '2023-09-01' AND '2024-05-31' THEN '2023'
+     END AS earliest_dec_suggested_true_cohort
+    ,CASE
+      WHEN pro.partnership_id IS NOT NULL AND pro.partnership_relationship = FALSE AND dec.lead_provider_name = 'National Institute of Teaching' THEN 1
+      ELSE 0
+     END AS niot_partnership_chk
+    ,ROW_NUMBER() OVER(PARTITION BY pro.participant_profile_id ORDER BY pro.participant_profile_id ASC) AS participant_precedence
+  FROM 
+    golden_thread pro
+  LEFT JOIN 
+    declarations dec 
+  ON
+    pro.participant_profile_id = dec.participant_profile_id
+  WHERE
+    pro.participant_role IS NOT NULL
+  ORDER BY
+    pro.user_id ASC, pro.participant_profile_id ASC, pro.participant_role ASC, dec.declaration_date ASC
+),
+
+-- Build preliminary output data which includes the binary result for cohort matching each of the suggested values.
+output_data AS (
+  SELECT 
+    *
+    ,CASE

```

Comment:
Could you do case when instr(start_date_suggested_true_cohort,cast(current_cohort as text))>0 and cover every case? That converts the cohort to a string and checks if it's in the text.